core/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -253,4 +396,110 @@ public Registration registerDispatchInterceptor(MessageDispatchInterceptor<Query
                             .map(queryHandler -> (MessageHandler<? super QueryMessage<?, ?>>) queryHandler)
                             .collect(Collectors.toList());
     }
+
+    private class SimpleQueryUpdateEmitter<U> implements QueryUpdateEmitter<U> {
+
+        private UpdateHandler<?, U> updateHandler;
+        private final AtomicReference<Runnable> registrationCanceledHandlerRef = new AtomicReference<>();
+        private volatile boolean active;
+        private final ReentrantReadWriteLock producerConsumerLock = new ReentrantReadWriteLock(true);
+        private final ReentrantLock initialLock;
+        private final Condition initialCondition;
+        private volatile boolean initial;
+
+        SimpleQueryUpdateEmitter(UpdateHandler<?, U> updateHandler, ReentrantLock initialLock,
```

Comment:
This class should create and manage its own locks, instead of relying on the `SimpleQueryBus` to create them.